File: src/main/java/dev/gustavpersson/thorincompanies/data_access_layer/entities/EmployeeEntity.java
package dev.gustavpersson.thorincompanies.data_access_layer.entities;

import com.j256.ormlite.field.DatabaseField;
import com.j256.ormlite.table.DatabaseTable;

import java.util.Date;

@DatabaseTable(tableName = "employees")
public class EmployeeEntity {

    @DatabaseField(generatedId = true)
    private int id;

    @DatabaseField(foreign = true)
    private CompanyEntity companyId;

    public CompanyEntity getCompanyId() {
        return companyId;
    }

    public void setCompanyId(CompanyEntity companyId) {
        this.companyId = companyId;
    }

    public Date getHiringDate() {
        return hiringDate;
    }

    public void setHiringDate(Date hiringDate) {
        this.hiringDate = hiringDate;
    }

    @DatabaseField()
    private Date hiringDate;

}

File: src/main/java/dev/gustavpersson/thorincompanies/data_access_layer/entities/EmployeeEntity.kt
package dev.gustavpersson.thorincompanies.data_access_layer.entities

import org.jetbrains.exposed.dao.IntEntity
import org.jetbrains.exposed.dao.IntEntityClass
import org.jetbrains.exposed.dao.id.EntityID
import org.jetbrains.exposed.dao.id.IntIdTable
import org.jetbrains.exposed.sql.javatime.date

object EmployeesTable: IntIdTable("thorin_employees") {
    val company = reference("company", CompaniesTable)
    val hiredDate = date("hiredDate")
}
class EmployeeEntity(id: EntityID<Int>): IntEntity(id) {
    companion object : IntEntityClass<EmployeeEntity>(EmployeesTable)
    var company by EmployeesTable.company
    var hiredDate by EmployeesTable.hiredDate
}
